lib/js/components/Buttons/IconButton/IconButton.vue

```diff
@@ -50,42 +57,83 @@
 @import '../../../../styles/settings/spacings';
 @import '../../../../styles/settings/typography';
 
-@mixin iconButtonColor($color, $hover-color: '') {
-	@if $hover-color == '' {
-		//$hover-color: mix($color, $color-firefly-black, (1 - $button-hover-alpha) * 100%);
-	}
-
-	color: $color;
-
-	&:hover,
-	&:active {
-		color: $hover-color;
-	}
-}
-
 .a-iconButton {
 	$self: &;
 
-	@each $color-name, $color-map in $theme-calculated-colors {
+	@each $color-name, $color-map in $icon-button-colors {
 		&.-color-#{$color-name} {
-			@include iconButtonColor(map-get($color-map, 'color'), map-get($color-map, 'hovered'));
-		}
-	}
+			#{$self}__label {
+				color: map-get($color-map, 'outlined', 'color');
+			}
 
-	@each $color-name, $color in $regular-colors {
-		&.-color-#{$color-name} {
-			@include iconButtonColor($color);
+			&:hover,
+			&.-hovered {
+				#{$self}__button {
```

Comment:
To check by me, is `background-color` and `color` necessary. Maybe it could be replaced by styling `__icon` instead of `__button`

Reply:
1. I think it's not worth to make changes. The problem is that class `hovered` is above `a-button` in the HTML tree. So if we want to move `__button` resposibility to `Button` it will more complicated.
2. Changing to the `__icon` will not fix it, because icon is inside. We need change background of the button (wrapper of icon). In above code we change `a-iconButton__button` element, not `a-button` component. It's the same HTML tag with classes:
```
class="a-button a-iconButton__button -outlined -color-primary"
```